gateway/src/apicast/upstream.lua

```diff
@@ -13,27 +13,34 @@ local str_format = string.format
 local resty_resolver = require('resty.resolver')
 local resty_url = require('resty.url')
 local core_base = require('resty.core.base')
+local http_proxy = require('apicast.http_proxy')
+local str_find = string.find
+local str_sub = string.sub
 local new_tab = core_base.new_tab
 
 local _M = {
 
 }
 
 local function proxy_pass(upstream)
-    local uri = upstream.uri
-
-    return str_format('%s://%s%s%s%s',
-            uri.scheme,
-            upstream.upstream_name,
-            uri.path or ngx.var.uri or '',
-            ngx.var.is_args or '',
-            ngx.var.query_string or '')
+    return str_format('%s://%s', upstream.uri.scheme, upstream.upstream_name)
 end
 
 local mt = {
     __index = _M
 }
 
+local function split_path(path)
+    if not path then return end
+
+    local start = str_find(path, '?', 1, true)
```

Comment:
Can the find + subs be replaced with a single "split by ?" ?

Reply:
I did some benchmarking, but I don't have the results and I think this was fully JIT compatible and faster.